silhouette/Graphtec.py

```diff
@@ -1036,29 +1142,32 @@ def plot(s, mediawidth=210.0, mediaheight=297.0, margintop=None,
     #p = b"FU%d,%d\x03" % (height,width) # optional
     #s.write(p)
 
-    if s.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3 or s.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO4:
+    if self.product_id() in [PRODUCT_ID_SILHOUETTE_CAMEO3, PRODUCT_ID_SILHOUETTE_CAMEO4]:
       pass
     else:
-      p = b"\\0,0\x03Z%d,%d\x03L0\x03FE0,0\x03FF0,0,0\x03" % (height, width)
-      s.write(p)
+      self.send_command([
+        "\\0,0",
+        "Z%d,%d" % (height, width),
+        "L0",
+        "FE0,0",
+        "FF0,0,0"])
 
     bbox['clip'] = {'urx':width, 'ury':top, 'llx':left, 'lly':height}
     bbox['only'] = bboxonly
-    cmd_list = s.plot_cmds(pathlist,bbox,offset[0],offset[1])
-    p = b'\x03'.join(cmd_list)
+    cmd_list = self.plot_cmds(pathlist,bbox,offset[0],offset[1])
 
     if bboxonly == True:
       # move the bounding box
-      p = b"M%d,%d" % (int(0.5+bbox['ury']), int(0.5+bbox['llx']))
-      p += b"\x03D%d,%d" % (int(0.5+bbox['ury']), int(0.5+bbox['urx']))
-      p += b"\x03D%d,%d" % (int(0.5+bbox['lly']), int(0.5+bbox['urx']))
-      p += b"\x03D%d,%d" % (int(0.5+bbox['lly']), int(0.5+bbox['llx']))
-      p += b"\x03D%d,%d" % (int(0.5+bbox['ury']), int(0.5+bbox['llx']))
-    p += b"\x03"   # Properly terminate string of plot commands.
+      cmd_list.append("M%d,%d" % (int(0.5+bbox['ury']), int(0.5+bbox['llx'])))
+      cmd_list.append("D%d,%d" % (int(0.5+bbox['ury']), int(0.5+bbox['urx'])))
+      cmd_list.append("D%d,%d" % (int(0.5+bbox['lly']), int(0.5+bbox['urx'])))
+      cmd_list.append("D%d,%d" % (int(0.5+bbox['lly']), int(0.5+bbox['llx'])))
+      cmd_list.append("D%d,%d" % (int(0.5+bbox['ury']), int(0.5+bbox['llx'])))
```

Comment:
The current implementation overrides `p` here, so as to discard the command list obtained with `self.plot_cmds`, as we apparently want to draw only the bbox. So I presume the first line should be `cmd_list =...` or an additional `cmd_list = []` would be needed.

Reply:
ok, I see it.